tests/10-api.spec.js

```diff
@@ -11,4 +12,83 @@ describe('API', () => {
     should.exist(m);
     m.memoize.should.be.a('function');
   });
+  describe('memoize', () => {
+    it('adds an item to the cache', async () => {
+      const m = new lruMemoize.default.LruMemoize();
+      let executedTestFn = false;
+      const testFn = async () => {
```

Comment:
because this is async it should return a promise:

```js
const foo = async () => 'foo';
const p = foo();
p
//Promise {<fulfilled>: "foo"}
const result = await foo();
result
// "foo"
```

so `testFn` is fine as is you don't need to wrap it again.

Reply:
Not sure what you're getting at here.  All `async` function return a promise regardless of how the return value is defined inside the function.